### DISCUSSION TITLE
Risk management, possible to call account equity?

### DISCUSSION THREAD

**Discussion by PlainStack** [kernc/backtesting.py#814] [Q&A]
I've been testing different strategies. Core to most of my strategies is to have very tight risk management, i.e. only risk 1% of account on every trade.

For this to work I need to change the size of each trade depending on my distance to the stop loss price. Is it possible to call the account equity in the next() function so I can calculate a variable order size for each trade? Below the variables and the size_long variable I'm trying to solve. I'm not sure if or how to call the account_equity.

        price = self.data.Close[-1]
        sl_short = self.data.Swing_High[-1]
        sl_long = self.data.Swing_Low[-1]
        profit_R = 1.5
        tp_long = price + ((price - sl_long) * profit_R)
        tp_short = price + ((price - sl_short) * profit_R)
        risk = 0.01
        size_long = (risk * account_equity) / (price / sl_long -1)

Feedback is much appreciated!

**Comment by PlainStack** [comment]
Okay so I tried the below which gives me an outcome that is relatively close, the rounding means it will be slightly more or less that 1% risked. The size _long runs fine but the size_short gives me OverflowError: cannot convert float infinity to integer. So not sure where I'm going wrong or if I'm even on the right track.

        #Risk management/ order sizing
        risk = 0.01
        account_equity = self.equity
        size_long = round(((risk * account_equity) / abs(price / sl_long -1)) / price)
        size_short = round(((risk * account_equity) / abs(price / sl_short -1)) / price)
        print(account_equity, size_short, price, sl_short)

print returns 1000000 144 22883.0 22952.5

Another thing I noticed is that the account equity is the original cash balance I used (1000000) and doesn't change with PnL changes.

**Reply by kernc** [reply]
OverflowError because you can't round Inf, probably because you're dividing by or something close to 0.

[`Strategy.equity`](https://kernc.github.io/backtesting.py/doc/backtesting/backtesting.html#backtesting.backtesting.Strategy.equity) is a property. In Python, `account_equity` holds the float value, not the reference.
https://github.com/kernc/backtesting.py/blob/65f54f6819cac5f36fd94ebf0377644c62b4ee3d/backtesting/backtesting.py#L753-L755

**Reply by PlainStack** [reply]
Thanks figured out what was wrong with the error, close == sl. So had to work around it.
